ARCs/arc-0014.md

```diff
@@ -0,0 +1,247 @@
+---
+arc: 14
+title: Authentication with Algorand accounts
+description: Use Algorand accounts to authenticate with third-party services
+author: Stefano De Angelis (@deanstef)
+discussions-to: https://github.com/algorandfoundation/ARCs/issues/42
+status: Draft
+type: Meta
+created: 2022-04-05
+---
+
+# Authentication with Algorand accounts
+
+A standard approach to authenticate users with Algorand accounts.
+
+## Abstract
+
+This ARC introduces a standard for SSO authentication based on Algorand accounts. It leverages the public-secret key <*PK, SK*> encryption schema to verify the identity of a user that owns an Algorand account. This approach fosters the adoption of novel identity management systems for Web3 applications.
+
+### Definitions
+
+- **System**: any frontend/backend application, service provider, or in general an entity not based on blockchain;
+- **Blockchain identity**: a public/secret key pair <*PK, SK*> representing a blockchain account;
+- **Algorand account**: a blockchain identity on Algorand identified with the key pair <*PKa, SKa*>;
+- **Algorand address**: the public key *PKa* of an Algorand account;
+- **User**: an Algorand account holder;
+- **Verifier**: a *system* that need to verify the identity of a User;
+- **dApp**: a decentralized Algorand application that natively runs on the Algorand blockchain, aka "*smart contract*",
+- **Wallet**: an off-chain application that stores the secret keys *SKa*s of Algorand accounts and can display and sign transactions for these accounts;
+- **message**: a generic string of bytes;
+- **digital signature**: a message signed with the private key of a blockchain identity, and specifically in this ARC with the *SKa*;
+
+## Motivation
+
+Traditional systems enforce SSO login leveraging on authentication of users with *credentials*, i.e. username and password. Once the identity of a user is verified, the system establishes an authenticated session. Sessions allows users to interact with the system without authenticating themselves for every interaction.
+
+> ARC-14 does not define any standard for session handling
+
+In a blockchain context, users do not have credentials. Conversely, in the Web3 users are identified through their unique blockchain public addresses (accounts for Algorand). Therefore, traditional authentication mechanisms result impractical. To interact with a dApp, users connect their wallet to the application. Using an authentication layer based on credentials on top of that mechanism might result redundant and inefficient.
```

Comment:
> In a blockchain context, users do not have credentials. 

An account pk/sk is a credential. The terminology needs updating to more clearly state something along the lines of "Here we present an alternative to username/password authentication using Algorand accounts"